lib/trailblazer/operation/inject.rb

```diff
@@ -19,14 +19,19 @@ def self.Defaults(default_dependencies)
       # @returns Task
       # @param Hash list of key/value that should be set if not already assigned/set before (or injected from the outside).
       class ReverseMergeDefaults
-        def initialize(defaults)
+        def initialize(defaults, allow_reset)
           @defaults = defaults
+          @allow_reset = allow_reset
         end
 
         def call((wrap_ctx, original_args), **circuit_options)
           ctx = original_args[0][0]
 
-          @defaults.each { |k, v| ctx[k] ||= v }
+          if @allow_reset
```

Comment:
I know that you don't like `if` but it feels necessary here or you suggest a better syntax?